openmetadata-service/src/main/java/org/openmetadata/service/search/SearchRepository.java

```diff
@@ -586,9 +586,12 @@ public void deleteOrUpdateChildren(EntityInterface entity, IndexMapping indexMap
             List.of(new ImmutablePair<>("service.id", docId)));
       }
       default -> {
-        searchClient.deleteEntityByFields(
-            indexMapping.getChildAliases(clusterAlias),
-            List.of(new ImmutablePair<>(entityType + ".id", docId)));
+        List<String> indexNames = indexMapping.getChildAliases(clusterAlias);
+        if (!indexNames.isEmpty()) {
+          searchClient.deleteEntityByFields(
+              indexMapping.getChildAliases(clusterAlias),
```

Comment:
can't we just pass the `indexNames` here (vs calling `indexMapping.getChildAliases(clusterAlias)` again?)

Reply:
yes @TeddyCr , did the change.  thanks!